pkg/registry/generic/registry/store_test.go

```diff
@@ -1091,17 +1130,19 @@ func TestStoreWatch(t *testing.T) {
 			}
 			wi.Stop()
 		}
-
+		destroyFunc()
 		server.Terminate(t)
 	}
 }
 
-func newTestGenericStoreRegistry(t *testing.T, hasCacheEnabled bool) (*etcdtesting.EtcdTestServer, *Store) {
+func newTestGenericStoreRegistry(t *testing.T, hasCacheEnabled bool) (*etcdtesting.EtcdTestServer, func(), *Store) {
 	podPrefix := "/pods"
 	server := etcdtesting.NewEtcdTestClientServer(t)
 	strategy := &testRESTStrategy{api.Scheme, api.SimpleNameGenerator, true, false, true}
+
 	codec := testapi.Default.StorageCodec()
 	s := etcdstorage.NewEtcdStorage(server.Client, codec, etcdtest.PathPrefix(), false, etcdtest.DeserializationCacheSize)
+	destroyFunc := func() {}
```

Comment:
Also found another dismissed leak here: https://github.com/kubernetes/kubernetes/blob/9fc0e1e98d791b79901ced941e52c87b70fba372/pkg/registry/pod/rest/log_test.go#L30

Gonna fix them in another PR too.